cachi2/core/package_managers/gomod.py

```diff
@@ -875,6 +882,12 @@ def _resolve_gomod(
         flags, app_dir, config.gomod_strict_vendor
     )
     if should_vendor:
+        if go_work_path and go_work_path.join_within_root("vendor").path.is_dir():
+            # NOTE: the same error will be reported even for 1.21 which doesn't support workspace
+            # vendoring, but given it's an invalid configuration and that we plan full 1.22 support
+            # in the foreseeable future, a not so user friendly error should be fine
+            raise UnsupportedFeature("Workspace vendoring introduced is not supported")
```

Comment:
nit: Consider rephrasing the error message slightly. Perhaps something like "Go workspace vendoring is not supported"